Cpp2IL.Decompiler/CilGenerator.cs

```
@@ -0,0 +1,346 @@
+                        Add(new CilInstruction(CilOpCodes.Ldc_I4, (int)low), source);
+                        Add(new CilInstruction(CilOpCodes.Conv_I8), source); // Convert low to long
+                        Add(new CilInstruction(CilOpCodes.Shl, 32), source); // Shift high left
+                        Add(new CilInstruction(CilOpCodes.Or), source); // Combine those
+                    }
+
+                    break;
+
+                case string text:
+                    Add(new CilInstruction(CilOpCodes.Ldstr, text), source);
+                    break;
+                case LocalVariable local:
+                    if (method.ParameterLocals.Contains(local))
+                    {
+                        var index = method.ParameterLocals.FindIndex(l => l == local);
+
+                        if (!method.Definition.IsStatic && index == 0)
+                            Add(new CilInstruction(CilOpCodes.Ldarg, method.Definition.Parameters.ThisParameter), source);
+                        else
+                            Add(new CilInstruction(CilOpCodes.Ldarg, method.Definition.Parameters[index]), source);
+
+                        break;
+                    }
+
+                    Add(new CilInstruction(CilOpCodes.Ldloc, GetLocal(local)), source);
+
+                    if (local.Field is { } field)
+                        Add(new CilInstruction(CilOpCodes.Ldfld, field.ImportWith(importer)), source);
+
+                    break;
+                case FieldDefinition staticField:
+                    Add(new CilInstruction(CilOpCodes.Ldfld, staticField.ImportWith(importer)), source);
+                    break;
+                case MemoryAddress memory:
+                    Add(new CilInstruction(CilOpCodes.Ldstr, memory.ToString()), source);
+                    break;
+                default:
+                    Add(new CilInstruction(CilOpCodes.Ldstr, $"unknown operand: {operand}"), source);
```

This is a silent failure. The only way we know it doesn't succeed is through manual inspection. I would prefer a more explicit failure.